packages/ra-ui-materialui/src/input/NumberInput.stories.tsx

```diff
@@ -26,6 +26,42 @@ export const Basic = () => (
     </AdminContext>
 );
 
+export const ReadOnly = ({ onSuccess = console.log }) => (
+    <AdminContext>
+        <Create
+            resource="posts"
+            record={{ id: 123, views: 23 }}
+            sx={{ width: 600 }}
+            mutationOptions={{ onSuccess }}
+        >
+            <SimpleForm>
+                <NumberInput source="views" readOnly />
+                <NumberInput source="price" readOnly />
+                <FormInspector name="views" />
+                <FormInspector name="price" />
+            </SimpleForm>
+        </Create>
+    </AdminContext>
+);
+
+export const Disabled = (onSuccess = console.log) => (
+    <AdminContext>
+        <Create
+            resource="posts"
+            record={{ id: 123, views: 23 }}
+            sx={{ width: 600 }}
+            mutationOptions={{ onSuccess }}
+        >
+            <SimpleForm>
+                <NumberInput source="views" disabled />
+                <NumberInput source="price" disabled />
+                <FormInspector name="views" />
+                <FormInspector name="price" />
```

Comment:
**polish**: there is not `price` in the record, I think this line is useless
```suggestion
                
```

Reply:
It's voluntary. It's a request from JB: to be able to see input behavior with and without data.